src/Storages/IStorage.h

```diff
@@ -168,6 +168,25 @@ class IStorage : public std::enable_shared_from_this<IStorage>, public TypePromo
         metadata.set(std::make_unique<StorageInMemoryMetadata>(metadata_));
     }
 
+    /// Daisy: start.
+    /// Get mutable version (snapshot) of storage CreateQuery. CreateQuery object is
+    /// multiversion, so it can be concurrently changed, but returned copy can be
+    /// used without any locks.
+    StorageInMemoryCreateQuery getInMemoryCreateQuery() const { return *create_query.get(); }
+
+    /// Get immutable version (snapshot) of storage CreateQuery. CreateQuery object is
+    /// multiversion, so it can be concurrently changed, but returned copy can be
+    /// used without any locks.
+    StorageInMemoryCreateQueryPtr getInMemoryCreateQueryPtr() const { return create_query.get(); }
+
+    /// Update storage CreateQuery. Used in ALTER or initialization of Storage.
+    /// CreateQuery object is multiversion, so this method can be called without
+    /// any locks.
+    void setInMemoryCreateQuery(const StorageInMemoryCreateQuery & create_query_)
```

Comment:
Can we make the signature as `void setInMemoryCreateQuery(std::unique<StorageInMemoryCreateQuery> create_query_)` ?

Reply:
Little difference, same above